client/signup/steps/site-type/form.jsx

```diff
@@ -25,6 +26,7 @@ class SiteTypeForm extends Component {
 	static propTypes = {
 		siteType: PropTypes.string,
 		submitForm: PropTypes.func.isRequired,
+		goToNextStep: PropTypes.func.isRequired,
```

Comment:
I find the distinction between the `goToNextStep` and `submitForm` prop pretty subtle. How about including `<ImportButton>` at the bottom of the `<SiteType>` component instead of `<SiteTypeForm>`? Visually I also think it sits outside the "form" but that's probably arguable.

It also fits in with my other comment about keeping flow switching logic in `index.jsx`